xtask/src/check_feature_dependencies.rs

```diff
@@ -63,6 +61,76 @@ pub fn check_feature_dependencies(shell: Shell, arguments: Arguments) -> anyhow:
         },
+        Requirement {
+            human_readable_name: "wasm32 with `webgl` feature depends on `glow`",
+            target: "wasm32-unknown-unknown",
+            packages: &["wgpu"],
+            features: &["webgl"],
+            default_features: false,
+            search_terms: &[Search::Positive("glow")],
+        },
+        Requirement {
+            human_readable_name: "apple with `vulkan` feature does not depend on `ash`",
+            target: "aarch64-apple-darwin",
+            packages: &["wgpu"],
+            features: &["vulkan"],
+            default_features: false,
+            search_terms: &[Search::Negative("ash")],
+        },
+        Requirement {
+            human_readable_name:
+                "apple with `vulkan-portability` feature depends on `ash` and `renderdoc-sys`",
+            target: "aarch64-apple-darwin",
+            packages: &["wgpu"],
+            features: &["vulkan-portability"],
+            default_features: false,
+            search_terms: &[Search::Positive("ash"), Search::Positive("renderdoc-sys")],
+        },
+        Requirement {
+            human_readable_name: "apple with 'gles' feature does not depend on 'glow'",
+            target: "aarch64-apple-darwin",
+            packages: &["wgpu"],
+            features: &["gles"],
+            default_features: false,
+            search_terms: &[Search::Negative("glow")],
+        },
+        Requirement {
+            human_readable_name: "apple with 'angle' feature depends on 'glow' and `renderdoc-sys`",
+            target: "aarch64-apple-darwin",
+            packages: &["wgpu"],
+            features: &["angle"],
+            default_features: false,
+            search_terms: &[Search::Positive("glow"), Search::Positive("renderdoc-sys")],
+        },
+        Requirement {
+            human_readable_name: "apple with no features does not depend on 'renderdoc-sys'",
+            target: "aarch64-apple-darwin",
+            packages: &["wgpu"],
+            features: &[],
+            default_features: false,
+            search_terms: &[Search::Negative("renderdoc-sys")],
+        },
+        Requirement {
+            human_readable_name:
+                "windows with no features does not depend on 'glow', `windows`, or `ash`",
+            target: "x86_64-pc-windows-msvc",
+            packages: &["wgpu"],
+            features: &[],
+            default_features: false,
+            search_terms: &[
+                Search::Negative("glow"),
+                Search::Negative("windows"),
+                Search::Negative("ash"),
+            ],
+        },
+        Requirement {
+            human_readable_name: "windows with no features depends on renderdoc-sys",
+            target: "x86_64-pc-windows-msvc",
+            packages: &["wgpu"],
+            features: &[],
+            default_features: false,
+            search_terms: &[Search::Positive("renderdoc-sys")],
```

Comment:
with no features we'd ideally not even depend on wgpu-core at all :/
But I reckon this is one of the drawbacks of this new solution - we can't opt out of wgpu-core as easily.

Not a blocker especially given that my exploration for doing this on the wgpu level turned out to be such a clusterf***

Reply:
> But I reckon this is one of the drawbacks of this new solution - we can't opt out of wgpu-core as easily.

There's not a ton of reason to depend on wgpu without wgpu-core anyway - maybe to allow types without `core`. Idk not really a concern I have now.